test/common/rds/rds_test.cc

```diff
@@ -175,9 +175,9 @@ virtual_hosts: null
     auto response =
         TestUtility::parseYaml<envoy::service::discovery::v3::DiscoveryResponse>(response_json);
     const auto decoded_resources = TestUtility::decodeResources<RouteConfiguration>(response);
-    EXPECT_TRUE(server_factory_context_.cluster_manager_.subscription_factory_.callbacks_
-                    ->onConfigUpdate(decoded_resources.refvec_, response.version_info())
-                    .ok());
+    THROW_IF_NOT_OK(
```

Comment:
Seems that there are 3 internal uses of this function. Consider changing the return type of this function (return the status), and EXPECT_EQ in the calling sites.

Reply:
I am super happy to change the signature but I think this preserves prior behavior.  Previously the bass class onConfigUpdate would throw on fail, bypass the EXPECT_TRUE and individual tests would EXPECT_THROW for errors, so I think this will make less work for folks changing tests.  lmk if you want the change with that context and I'm on it though

Reply:
Apologies, I think I missed that there may be other reasons where the onConfigUpdate may throw an exception, and so changing the call-sites to EXPECT_EQ won't work.